saleor/core/permissions.py

```diff
@@ -157,3 +157,26 @@ def get_permissions_from_codenames(permission_codenames: List[str]):
         .prefetch_related("content_type")
         .order_by("codename")
     )
+
+
+def _permission_required(perms, context):
+    from saleor.account.models import User
+
+    if isinstance(context, User):
```

Comment:
Wouldn't be better (instead of this `if/else`?:
```
if isinsance(requestor, "App") and AccountPermissions.MANAGE_STAFF in perms:
    return False
else:
    return requestor.has_perms(perms)
```